qemu/tests/fio_perf.py

```diff
@@ -0,0 +1,259 @@
+        """
+        pin guest vcpu and vhost threads to cpus of a numa node repectively
+
+        :param vm: VM object
+        :param node: which numa node to pin
+        """
+        if node:
+            if not isinstance(node, utils_misc.NumaNode):
+                node = utils_misc.NumaNode(int(node))
+            utils_test.qemu.pin_vm_threads(vm, node)
+
+        return node
+
+    # login virtual machine
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    login_timeout = int(params.get("login_timeout", 360))
+    session = vm.wait_for_login(timeout=login_timeout)
+    logging.debug(process.system_output("numactl --hardware"))
+    logging.debug(process.system_output("numactl --show"))
+    _pin_vm_threads(vm, params.get("numa_node"))
+
+    # get parameter from dictionary
+    fio_cmd = params["fio_cmd"]
+    rw = params["rw"]
+    block_size = params["block_size"]
+    iodepth = params["iodepth"]
+    threads = params["threads"]
+    cmd_timeout = int(params.get("cmd_timeout", 1200))
+    order_list = params["order_list"]
+    driver_format = params.get("drive_format")
+    guest_ver_cmd = params["guest_ver_cmd"]
+    pattern = params["pattern"]
+    pre_cmd = params["pre_cmd"]
+    format = params.get("format")
+    md5value = params.get("md5value")
+    compile_cmd = params.get("compile_cmd")
+    test_src = params.get("test_src")
+    os_type = params.get("os_type", "linux")
+    image_format = params.get("image_format")
+    drop_cache = params.get("drop_cache")
+    backend_type = params.get("backend_type")
+
+    result_path = utils_misc.get_path(test.resultsdir,
+                                      "fio_result.RHS")
+    result_file = open(result_path, "w")
+
+    # scratch host and windows guest version info
+    get_version(session, result_file, guest_ver_cmd, os_type,
+                driver_format, cmd_timeout)
+
+    # online disk
+    if os_type == "windows":
+        for num in range(1, 2):
+            disks = check_status(session, cmd_timeout, num)
+            diskstatus = re.findall("Disk\s+\d+\s+(\w+).*?\s+\d+", disks[0])[0]
+            if diskstatus == "Offline":
+                online_disk_cmd = params.get("online_disk_cmd")
+                online_disk_run = online_disk_cmd % num
+                (s, o) = session.cmd_status_output(online_disk_run,
+                                                   timeout=cmd_timeout)
+                if s:
+                    raise exceptions.TestFail("Failed to online disk: %s" % o)
+    # format disk
+    if format == "True":
+        session.cmd(pre_cmd, cmd_timeout)
+
+    # get order_list
+    order_line = ""
+    for order in order_list.split():
+        order_line += "%s|" % format_result(order)
+
+    # get result tested by each seanario
+    for io_pattern in rw.split():
+        result_file.write("Category:%s\n" % io_pattern)
+        result_file.write("%s\n" % order_line.rstrip("|"))
+        if io_pattern == "rewrite":
```

Comment:
We rarely use "rewrite" in our general test, so when we use it we will temporarily modify the cfg in local machine. If it confuses others, I will consider to remove it on code.